app/services/purchase/create_service.rb

```diff
@@ -51,6 +51,7 @@ def perform
         )
       elsif @product.product_refund_policy_enabled?
         purchase.build_purchase_refund_policy(
+          max_refund_period_in_days: @product.product_refund_policy.max_refund_period_in_days,
```

Comment:
The main change that allows us saving the value to `PurchaseRefundPolicy` records moving forward.